### PR TITLE
Introduce ScalingParameters for Scalable

### PR DESCRIPTION
**Pull request by colinepiloquet**
Signed-off-by: Coline PILOQUET <coline.piloquet@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix: adds a boolean saying whether the terminal should be reconnected in case of scaling on a disconnected element.

**What is the current behavior?** *(You can also link to an open issue here)*
The terminal is always reconnected.

### CODE REVIEW

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/AbstractCompoundScalable.java [hunk 1]
```diff
@@ -71,9 +71,14 @@ public double minimumValue(Network n, ScalingConvention powerConvention) {
...
     }
 
+    @Override
+    public double scale(Network n, double asked, boolean reconnect) {
+        return scale(n, asked, ScalingConvention.GENERATOR, reconnect);
+    }
+
     @Override
     public double scale(Network n, double asked) {
-        return scale(n, asked, ScalingConvention.GENERATOR);
```
Comment (annetill) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/AbstractCompoundScalable.java:76]: I don't get this change: I think you must keep the convention and add a default behavior for reconnection. @phiedw I think that reconnection should not be the default behavior, what is you opinion? Why do you need a reconnection in capacity calculation?

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/DanglingLineScalable.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ public void filterInjections(Network n, List<Injection> injections, List<String>
      * If scalingConvention is GENERATOR, the load active power decreases for positive "asked" and increases inversely
      */
     @Override
-    public double scale(Network n, double asked, Scalable.ScalingConvention scalingConvention) {
+    public double scale(Network n, double asked, Scalable.ScalingConvention scalingConvention, boolean reconnect) {
```
Comment (geofjamg) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/DanglingLineScalable.java:112]: We really should consider grouping all theses parameters in a ScalingContext object.

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/DanglingLineScalable.java [hunk 3]
```diff
@@ -109,9 +109,9 @@ public void filterInjections(Network n, List<Injection> injections, List<String>
      * If scalingConvention is GENERATOR, the load active power decreases for positive "asked" and increases inversely
      */
     @Override
-    public double scale(Network n, double asked, Scalable.ScalingConvention scalingConvention) {
+    public double scale(Network n, double asked, ScalingContext context) {
```
Comment (geofjamg) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/DanglingLineScalable.java:112]: If we just add read only data in `ScalingContext`, it is not really a context. It is more a `ScalingParameter`

**Review by geofjamg — APPROVED** [review]

**Review by miovd — APPROVED** [review]

**Review by flo-dup — APPROVED** [review]
Nice, just a tiny javadoc comment!

**Review by phiedw — APPROVED** [review]

**Pull request merged**